src/v/wasm/transform_module.cc

```diff
@@ -203,11 +203,9 @@ void transform_module::start() {
 void transform_module::stop(const std::exception_ptr& ex) {
     if (_guest_cond_var) {
         _guest_cond_var->broken(ex);
-        _guest_cond_var = std::nullopt;
     }
     if (_host_cond_var) {
         _host_cond_var->broken(ex);
-        _host_cond_var = std::nullopt;
```

Comment:
QQ - Was the issue introduced by this PR or a previous one?

Reply:
It was introduced by the first PR - I'm assuming a suspension point changed somehow. If you look at the implementation of `broken` it sets a variable about which exception is to be thrown when the promise resolves, so if a waiter wasn't waiting on the condvar when `stop` is called it gets the wrong exception type than what the tests expect (they expect a wasm_exception not broken_condvar).

Reply:
Also it was only in release builds which (I only really test debug builds locally) and they have different behavior WRT suspension points & task reordering.